From ca22cd85093774a34e01ab4ceda5e145f15bbe2a Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Tue, 22 Mar 2022 14:22:37 +0200
Subject: [PATCH] Revert "Revert "[SW-43475] greco: arc: move preboot load address to start of sram""

This reverts commit 489bc16ab115357c3c2de8ecb1b02664d842ca8d.
This patch  reverts the revert of patch moving preboot texbase to start
of sram. Issue has been fixed in uboot and original patch is resubmited.

Change-Id: Iecd8867d7a15a15c48a34ab0790cbc20cd25c5de
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/greco/arc/common.h b/arch/greco/arc/common.h
index be0b07d..f83b41e 100644
--- a/arch/greco/arc/common.h
+++ b/arch/greco/arc/common.h
@@ -21,7 +21,7 @@
 #define SP_SRAM_BASE		0x1BFE0000
 #define ARC_ICCM_BASE		0x1B020000
 #define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + 0xC)
-#define PREBOOT_TEXT_BASE	(SDRAM_BASE + PPBOOT_SIZE + 0x8)
+#define PREBOOT_TEXT_BASE	(SDRAM_BASE + 0x8)
 
 #define HL_SP_SRAM_BASE		0x7FFBFE0000
 #define HL_SP_SRAM_SIZE		SZ_64K
diff --git a/arch/greco/arc_elf32/common.c b/arch/greco/arc_elf32/common.c
index f243644..31b0f9f 100644
--- a/arch/greco/arc_elf32/common.c
+++ b/arch/greco/arc_elf32/common.c
@@ -226,8 +226,7 @@
 	 * 0x7FFBFE1000ULL, the address to be passed is 0x7FFBFE0400ULL
 	 * instead of 0x7FFBFE1000ULL (right shift offest 0x1000 by 2).
 	 */
-	dma_copy_image(image_size, flash_offset,
-			(SDRAM_FULL_BASE + (PPBOOT_SIZE >> 2)),
+	dma_copy_image(image_size, flash_offset, SDRAM_FULL_BASE,
 			SPI_DMA_DST_HBW);
 	if (dma_wait_finish() == DMA_CRC_ERROR)
 		return ARCBOOT_BRINGUP_ERROR;
diff --git a/arch/greco/arc_elf32/common.h b/arch/greco/arc_elf32/common.h
index 139d9ad..ada859f 100644
--- a/arch/greco/arc_elf32/common.h
+++ b/arch/greco/arc_elf32/common.h
@@ -24,7 +24,7 @@
 #define SP_SRAM_BASE		0x1BFE0000
 #define HL_SP_SRAM_SIZE		SZ_64K
 #define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + 0xC)
-#define PREBOOT_TEXT_BASE	(SDRAM_BASE + PPBOOT_SIZE + 0x8)
+#define PREBOOT_TEXT_BASE	(SDRAM_BASE + 0x8)
 
 #define FW_WORD_NUM		0x2A00	/* FW_SIZE / 2 */
 
